src/transformStateWithClones.js

```diff
@@ -7,7 +7,89 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const actionTypes = [];
+  let REPEAT = false;
+
+  for (const action of actions) {
+    if (!actionTypes.includes(action.type)) {
+      actionTypes.push(action.type);
+    }
+  }
+
+  if (actionTypes.length === 1) {
+    REPEAT = true;
+  }
```

Comment:
The `actionTypes` array and the `REPEAT` flag are unnecessary. The purpose of the function is to apply each action to the state and push the modified state to `stateHistory` after each action. The current implementation with `REPEAT` complicates the logic and is not required by the task description.